app/src/main/java/org/schabi/newpipe/util/potoken/PoTokenGenerator.kt

```diff
@@ -0,0 +1,35 @@
+package org.schabi.newpipe.util.potoken
+
+import android.content.Context
+import io.reactivex.rxjava3.core.Single
+import java.io.Closeable
+
+/**
+ * This interface was created to allow for multiple methods to generate poTokens in the future (e.g.
+ * via WebView and via a local DOM implementation)
+ */
+interface PoTokenGenerator : Closeable {
+    /**
+     * Generates a poToken for the provided identifier, using the `integrityToken` and
+     * `webPoSignalOutput` previously obtained in the initialization of [PoTokenWebView]. Can be
+     * called multiple times.
+     */
+    fun generatePoToken(identifier: String): Single<String>
```

Comment:
Couldn't this be written as a suspend function instead? A `runBlocking` version could also be added to call this from Java.

Reply:
I guess we can change it to coroutines when we convert this to the refactor branch, but I don't think we have coroutines anywhere in NewPipe non-refactor, so I didn't want to introduce new libraries and new ways of writing code, so I just stuck to RxJava.